src/DotVVM.Framework/Compilation/CompiledAssemblyCache.cs

```diff
@@ -11,6 +24,79 @@ public class CompiledAssemblyCache
         private readonly ConcurrentDictionary<Assembly, MetadataReference> cachedAssemblyMetadata = new ConcurrentDictionary<Assembly, MetadataReference>();
         private readonly ConcurrentDictionary<string, Assembly> cachedAssemblies = new ConcurrentDictionary<string, Assembly>();
 
+        private readonly DotvvmConfiguration configuration;
+
+        public static CompiledAssemblyCache? Instance { get; private set; }
+
+        public CompiledAssemblyCache(DotvvmConfiguration configuration)
+        {
+            this.configuration = configuration;
+
+#if DotNetCore
+            System.Runtime.Loader.AssemblyLoadContext.Default.Resolving += DefaultOnResolving;
+#else
+            AppDomain.CurrentDomain.AssemblyResolve += TryResolveAssembly;
+#endif
+
+            foreach (var assembly in BuildReferencedAssembliesCache())
+            {
+                GetAssemblyMetadata(assembly);
+            }
+
+            if (Instance == null)
+            {
+                // normally, the constructor is not called multiple times as this service is singleton; only in unit tests, we create it multiple times
+                Instance = this;
+            }
+        }
+
+        private IEnumerable<Assembly> BuildReferencedAssembliesCache()
+        {
+            var diAssembly = typeof(ServiceCollection).Assembly;
+
+            var references = diAssembly.GetReferencedAssemblies().Select(Assembly.Load)
+                .Concat(configuration.Markup.Assemblies.Select(e => Assembly.Load(new AssemblyName(e))))
+                .Concat(new[] {
+                    diAssembly,
+                    Assembly.Load(new AssemblyName("mscorlib")),
+                    Assembly.Load(new AssemblyName("System.ValueTuple")),
+                    typeof(IServiceProvider).Assembly,
+                    typeof(RuntimeBinderException).Assembly,
+                    typeof(DynamicAttribute).Assembly,
+                    typeof(DotvvmConfiguration).Assembly,
+#if DotNetCore
+                    Assembly.Load(new AssemblyName("System.Runtime")),
+                    Assembly.Load(new AssemblyName("System.Collections.Concurrent")),
+                    Assembly.Load(new AssemblyName("System.Collections")),
+#else
+                    typeof(List<>).Assembly
+#endif
+                });
+
+            var netstandardAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == "netstandard");
```

Comment:
Hmm, so what is the point when we get all assemblies anyway here? Oo

Reply:
`GetAssemblies` doesn't load the assemblies, but it looks like that this line is not needed anyway.